index.js

```diff
@@ -142,16 +141,34 @@ function replaceEnvVars(config) {
     });
 }
 
+function requiresConfig(options) {
+    return !options.generate && (options.start || options.test);
+}
+
 function main() {
     const options = getOptions();
-    let configFile = options.configFile;
-    if (!/^\//.test(configFile)) {
+
+    let configFilePath = options.configFilePath || 'config.yaml';
+    if (!/^\//.test(configFilePath)) {
         // resolve relative paths
-        configFile = path.resolve(`${process.cwd()}/${configFile}`);
+        configFilePath = path.resolve(`${process.cwd()}/${configFilePath}`);
+    }
+
+    if (requiresConfig(options) && !fs.existsSync(configFilePath)) {
+        console.error('Service config is required but not provided');
+        process.exit(1);
+    }
+
+    let config;
```

Comment:
Not quite, we use `!config` in the docker script to verify it's provided when it's needed. Having a `TypeError` inn case some of the verification is incorrect is better then getting `undefined` for the property and silently doing something wrong.